### PR TITLE
[GEOT-6198] Do not set ROIs in image mosaic when not strictly necessary

### PR DESCRIPTION
**Pull request by aaime**
Depends on https://github.com/geosolutions-it/jai-ext/pull/227 and a jai-ext release

### CODE REVIEW

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/granulecollector/BaseSubmosaicProducer.java [hunk 1]
```diff
@@ -83,6 +83,8 @@
 
     protected Color inputTransparentColor;
 
+    // boolean forceFootprints = JAIExt.;
```
Comment (dromagnoli) [inline modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/granulecollector/BaseSubmosaicProducer.java]: Is it here for "debugging/testing" purposes?

File: pom.xml [hunk 2]
```diff
@@ -97,7 +97,7 @@
     <src.output>${basedir}/target</src.output>
     <imageio.ext.version>1.2.0</imageio.ext.version>
     <jts.version>1.16.0</jts.version>
-    <jaiext.version>1.1.4</jaiext.version>
+    <jaiext.version>1.1-SNAPSHOT</jaiext.version>
```
Comment (dromagnoli) [inline pom.xml]: I think you set it to SNAPSHOT for development and testing. should it be updated to 1.1.next once Jai-ext get released?

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/granulecollector/BaseSubmosaicProducer.java (hunk 1 above)
Comment (aaime) [reply modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/granulecollector/BaseSubmosaicProducer.java]: Yep, must be removed

File: pom.xml (hunk 2 above)
Comment (aaime) [reply pom.xml]: That's why I said "Depends on geosolutions-it/jai-ext#227 **and a jai-ext release**" :-D

**Comment by aaime** [comment]
Travis builds keep on failing for network related issues, merging anyways and waiting for the official build server verdict.

**Pull request merged**